lib/rubocop/cop/naming/method_name.rb

```diff
@@ -48,19 +70,57 @@ def on_send(node)
             name = attr_name(name_item)
             next if !name || matches_allowed_pattern?(name)
 
-            check_name(node, name, range_position(node))
+            if forbidden_name?(name.to_s)
+              register_forbidden_name(node)
+            else
+              check_name(node, name, range_position(node))
+            end
           end
         end
 
         def on_def(node)
           return if node.operator_method? || matches_allowed_pattern?(node.method_name)
 
-          check_name(node, node.method_name, node.loc.name)
+          if forbidden_name?(node.method_name.to_s)
+            register_forbidden_name(node)
+          else
+            check_name(node, node.method_name, node.loc.name)
+          end
         end
         alias on_defs on_def
 
         private
 
+        def forbidden_identifiers
+          cop_config.fetch('ForbiddenIdentifiers', [])
+        end
+
+        def forbidden_patterns
+          cop_config.fetch('ForbiddenPatterns', [])
+        end
+
+        def matches_forbidden_pattern?(name)
+          forbidden_patterns.any? { |pattern| Regexp.new(pattern).match?(name) }
+        end
+
+        def forbidden_name?(name)
+          (forbidden_identifiers.any? && forbidden_identifiers.include?(name)) ||
+            (forbidden_patterns.any? && matches_forbidden_pattern?(name))
+        end
```

Comment:
Since this is now essentially shared with the same methods in `Naming/VariableName`, we could extract a mixin.